raster/src/main/scala/geotrellis/raster/ByteArrayTile.scala

```diff
@@ -44,7 +44,8 @@ object ByteArrayTile {
         val v = bytes(i)
         if(v == replaceNoData)
           arr(i) = byteNODATA
-        arr(i) = bytes(i)
+        else
+          arr(i) = bytes(i)
```

Comment:
Could be

```scala
arr(i) = v
```
That will save 1/1000th of a millisecond most likely ;)

Also, the bug this fixes is awful and hilarious. Thanks for catching that.